### ISSUE TITLE
Current Iris normalization seems to worsen performance

### ISSUE THREAD

**Issue by andrenmelo** [Qiskit/qiskit-machine-learning#275]
### Environment

- **Qiskit Machine Learning version**: 0.3.0
- **Python version**: 3.9.7 

### What is happening?

The data normalization procedure in the `iris` dataset seems to make the classification harder. I came across this issue when trying to reproduce the results of [arXiv:2011.00027](https://arxiv.org/pdf/2011.00027.pdf), where a standard QNN is trained on the first two classes of iris with full batch size.

### How can we reproduce the issue?

Below I give an example reproducing the performance drop. I tried changing optimizers/optimizer parameters, but the issue persisted.

```python
from qiskit import Aer
from qiskit.algorithms.optimizers import SPSA
from qiskit.circuit.library import RealAmplitudes, ZZFeatureMap
from qiskit.utils import QuantumInstance

import qiskit_machine_learning
from qiskit_machine_learning import datasets
from qiskit_machine_learning.algorithms import VQC

import matplotlib.pyplot as plt
import numpy as np
from sklearn.datasets import load_iris
from sklearn import preprocessing

qi_statevector = QuantumInstance(Aer.get_backend('statevector_simulator'))

# Data normalized with qiskit machine learning
x = datasets.iris(300, 1, 4)
X_QISKIT = x[0][:99]
Y_QISKIT = x[1][:99, :2]

# Data as normalized by the authors of arXiv:2011.00027
# see also https://github.com/amyami187/effective_dimension/blob/master/Loss_plots/generate_data/hard_qnn.py
iris = load_iris()
x = preprocessing.normalize(iris.data)
x1_train = x[0:49, :] # class A
y1_train = np.array([[1, 0] for i in range(49)])
x2_train = x[50:99, :] # class B
y2_train = np.array([[0, 1] for i in range(49)])

X = np.vstack([x1_train, x2_train])
Y = np.vstack([y1_train, y2_train])

losses = []

def callback(weights, loss):
    print(loss)
    losses.append(loss)

model = VQC(4,
            ZZFeatureMap(4, 2), RealAmplitudes(4),
            optimizer=SPSA(),
            quantum_instance=qi_statevector,
            callback=callback)

# model.fit(X, Y) fits to 1.0 accuracy and loss ≈ 20
# model.fit(X_QISKIT, Y_QISKIT) fits to 0.8 accuracy and loss ≈ 70
```

### What should happen?

Iris is an "easy" dataset, so a simple QNN should be able to fit it well.

**Comment by adekusar-drl** [comment]
Hi, the `iris` function in Qiskit Machine Learning is not the same in `scikit-learn`. In Qiskit Machine Learning, the dataset functions like `iris`, `digits`, etc take a parameter `n` that describes how many features must be in the output dataset. Internally these function apply PCA to reduce the number of features to the desirable number defined by `n`. In this case despite in the original iris dataset there are 4 features and the same number 4 passed is passed as `n` to the QML's function `iris`, PCA is still applied to the dataset. And PCA transforms the dataset despite it is not really required. And the output dataset is different compared to the original one due to this transformation. And this is the reason why you get a different score. You fit models using different datasets. Is it a feature or bug? Well, a good question. Perhaps, if `n` is equal to the number of features in the dataset PCA should not be applied. But in the PCA implementation there's no such check and the output dataset is different to the input one. So, I'd say it is a feature. 

I hope this clarifies the problem.

**Comment by andrenmelo** [comment]
Dear @adekusar-drl: I understand the motivation behind the preprocessing done on qiskit machine learning. However, the current pipeline degrades performance at all n, not just 4. That to me indicates that the preprocessing should be done differently.

**Comment by adekusar-drl** [comment]
@andrenmelo Dimension reduction causes performance degradation, it is a no brainer for me. Do you have suggestion what can be changed?

**Comment by adekusar-drl** [comment]
I'm closing the issue since we are deprecating scikit learn datasets in #319.
